amule-gui: Does not compile on unstable ( CMake 4 issue )

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

amule-gui will fail to compile on nixos-unstable.

### Steps to reproduce

1. `nix shell github:NixOS/nixpkgs/nixos-unstable#amule-gui`

### Expected behaviour

Successful compilation

### Screenshots

_No response_

### Relevant log output

```console
CMake Error at CMakeLists.txt:1 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 3.5 has been removed from CMake.
```

### Additional context

[Full build logs](https://github.com/user-attachments/files/22977637/amule-gui.txt)

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.52, NixOS, 25.11 (Xantusia), 25.11.20251015.544961d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos, nixos-hardware"`
 - nixpkgs: `/nix/store/rcsmdj802sk3w9l8m0b0bvdy0jiw2wgr-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/am/amule/package.nix
+++ b/pkgs/by-name/am/amule/package.nix
@@ -79,6 +79,9 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     echo "find_package(Threads)" >> cmake/options.cmake
+
+    substituteInPlace src/libs/ec/abstracts/CMakeLists.txt \
+      --replace-fail "CMAKE_MINIMUM_REQUIRED (VERSION 2.8)" "CMAKE_MINIMUM_REQUIRED (VERSION 3.10)"
   '';
 
   # aMule will try to `dlopen' libupnp and libixml, so help it

